File: deleting an element in array.cpp
#include<iostream>
using namespace std;
int main()
{
    int x;
    int arr[8];
    cout<<"Enter the elements"<<endl;
    for(int i=0;i<8;i++)
    {
        cin>>arr[i];
    }
    cout<<"Enter the element you want to delete"<<endl;
    cin>>x;
    for(int i=0;i<8;i++)
    {
        if(arr[i]==x)
        {
            arr[i]=-1;
            break;
        }
        else
        {
            cout<<"Element not found"<<endl;
        }
    }
    cout<<endl;
    for(int i=0;i<8;i++)
    {
        if(arr[i]!=-1)
        {
            cout<<arr[i]<<endl;
        }
    }
}

File: deleting an element in array.java
import java.util.Scanner;
public class Delete
{
    public static void main(String a[])
    {
        int x;
        int[] arr = new int[5];
        Scanner s = new Scanner(System.in);
        System.out.println("Enter the elements");
        for(int i=0;i<5;i++)
        {
            arr[i]=s.nextInt();
        }
        System.out.println("Enter the element that you want to delete");
        x=s.nextInt();
        for(int i=0;i<5;i++)
        {
            if(arr[i]==x)
            {
                System.out.println("Element found at place "+i);
                arr[i]=-1;
                break;
            }
            else
            {
                System.out.println("Element is not present at "+i);
            }
        }
        for(int i=0;i<5;i++)
        {
            if(arr[i]!=-1)
            {
                System.out.println(arr[i]);
            }
        }

    }
}